examples/long-running-process/long-running.js

```diff
@@ -0,0 +1,125 @@
+                 * we may get an inactive unit here; watch for state changes. This will also update
+                 * the UI if the unit stops. */
+                this.subscription = systemd_client.subscribe(
+                    { interface: I_DBUS_PROP, member: "PropertiesChanged" },
+                    (path, iface, signal, args) => {
+                        if (path === unitObj && args[1].ActiveState)
+                            checkState(unit);
+                    });
+
+                systemd_client.call(unitObj, I_DBUS_PROP, "GetAll", [I_SD_UNIT], { type: "s" })
+                        .then(([props]) => {
+                            if (props.ActiveState.v === 'activating') {
+                                state.innerHTML = cockpit.format("$0 is running", unit);
+                                run_button.setAttribute("disabled", "");
+                                // StateChangeTimestamp property is in µs since epoch, but journalctl expects seconds
+                                showJournal(unit, "--since=@" + Math.floor(props.StateChangeTimestamp.v / 1000000));
+                            } else if (props.ActiveState.v === 'failed') {
+                                state.innerHTML = cockpit.format("$0 is not running and failed", unit);
+                                run_button.setAttribute("disabled", "");
+                                // Show the whole journal of this boot; this may be refined a bit with props.InvocationID
+                                showJournal(unit, "--boot");
+                            } else {
+                                /* Type=oneshot transient units only have state "activating" or "failed",
+                                 * or don't exist at all (handled below in NoSuchUnit case).
+                                 * If you don't care about "failed", call systemd-run with --collect */
+                                state.innerHTML = cockpit.format("Error: unexpected state of $0: $1", unit, props.ActiveState.v);
+                            }
+                        })
+                        .catch(error);
+            })
+            .catch(ex => {
+                if (ex.name === "org.freedesktop.systemd1.NoSuchUnit") {
+                    if (this.subscription) {
+                        this.subscription.remove();
+                        this.subscription = null;
+                    }
+                    state.innerHTML = cockpit.format("$0 is not running", unit);
+                    run_button.removeAttribute("disabled");
+                } else {
+                    error(ex);
+                }
+            });
+}
+
+/* Start long-running process, called on clicking the "Start Process" button.
+ * This runs as root, thus will be shared with all privileged Cockpit sessions. If the process
+ * should run with user privileges instead, drop the `superuser` and call `systemd-run` with `--user`.
```

Comment:
It's more complicated than "just add --user" here.  The journal command would need to be adjusted, and (probably) also the systemd1 instance you connect to.

It's fine to just say "this runs stuff as root" and not contemplate the user possibility.  If you do want to talk about the user one, it should probably get a more complete treatment.

Reply:
Indeed, that's incomplete. It would change quite a lot of code. If we ever get this use case (running as root is the much more common one), I'd rather suggest to add a completely separate example with its own test. By then we also have a library for this.

I entirely removed that comment.